DVLSForLinux/install-dvls.ps1

```diff
@@ -0,0 +1,435 @@
+
+if ($DVLSVariables.CreateDatabase) {
+    New-DPSDatabase -ConnectionString $Settings.ConnectionStrings.LocalSqlServer
+}
+
+Update-DPSDatabase -ConnectionString $Settings.ConnectionStrings.LocalSqlServer -InstallationPath $DVLSVariables.DVLSPath
+
+New-DPSDataSourceSettings -ConnectionString $Settings.ConnectionStrings.LocalSqlServer
+New-DPSEncryptConfiguration -ApplicationPath $DVLSVariables.DVLSPath
+New-DPSDatabaseAppSettings -Configuration $Configuration
+New-DPSAdministrator -ConnectionString $Settings.ConnectionStrings.LocalSqlServer -Name $DVLSVariables.DVLSAdminUsername -Password $DVLSVariables.DVLSAdminPassword -Email $DVLSVariables.DVLSAdminEmail
+
+if ($DVLSVariables.DVLSCertificate) {
+    Write-Host ("[{0}] Generating self-signed certificate" -F (Get-Date -Format "yyyyMMddHHmmss")) -ForegroundColor Green
+
+    & openssl req -x509 -newkey rsa:4096 -sha256 -days 3650 -nodes -keyout cert.key -out cert.crt -subj ("/CN={0}" -F $DVLSVariables.DVLSHostName) -addext ("subjectAltName=IP:{0}" -F $DVLSVariables.DVLSHostName) > /dev/null 2>&1
+    & openssl pkcs12 -export -out cert.pfx -inkey cert.key -in cert.crt -passout pass: > /dev/null 2>&1
+
+    $JSON = Get-Content -Path (Join-Path -Path $DVLSVariables.DVLSPath -ChildPath "appsettings.json") | ConvertFrom-Json -Depth 100
+
+    $JSON.Kestrel.Endpoints.Http | Add-Member -MemberType NoteProperty -Name "Certificate" -Value @{
+        "Path"     = (Join-Path -Path $DVLSVariables.DVLSPath -ChildPath "cert.pfx")
+        "Password" = ""
+    }
+
+    $JSON | ConvertTo-Json -Depth 100 | Set-Content -Path (Join-Path -Path $DVLSVariables.DVLSPath -ChildPath "appsettings.json")
+
+    Set-DPSAccessUri -ApplicationPath $DVLSVariables.DVLSPath -ConnectionString $Settings.ConnectionStrings.LocalSqlServer -AccessURI ("https://{0}:5000/" -F $DVLSVariables.DVLSHostName)
+
+    & sudo $PwshExecutable -Command {
+        Param(
+            $DVLSVariables
+        )
+        
+        & chown ("{0}:{1}" -F $DVLSVariables.DVLSUser, $DVLSVariables.DVLSGroup) (Join-Path -Path $DVLSVariables.DVLSPath -ChildPath "cert.pfx")
+        & chown ("{0}:{1}" -F $DVLSVariables.DVLSUser, $DVLSVariables.DVLSGroup) (Join-Path -Path $DVLSVariables.DVLSPath -ChildPath "cert.crt")
+        & chown ("{0}:{1}" -F $DVLSVariables.DVLSUser, $DVLSVariables.DVLSGroup) (Join-Path -Path $DVLSVariables.DVLSPath -ChildPath "cert.key")
+    } -Args $DVLSVariables
+}
+
+Write-Host ("[{0}] Installing systemd unit file to '{1}'" -F (Get-Date -Format "yyyyMMddHHmmss"), $DVLSVariables.SystemDPath) -ForegroundColor Green
+
+$SystemDTemplate = @"
+[Unit]
+Description=DVLS
+[Service]
+Type=simple
+Restart=always
+RestartSec=10
+User=$($DVLSVariables.DVLSUser)
+ExecStart=$($DVLSVariables.DVLSExecutable)
+WorkingDirectory=$($DVLSVariables.DVLSPath)
+KillSignal=SIGINT
+SyslogIdentifier=dvls
+[Install]
+WantedBy=multi-user.target
+Alias=dvls.service
+"@
+
+& sudo $PwshExecutable -Command {
```

Comment:
I haven't tested the install script, but it might be worth to check the permissions for the service once created to make sure only root can modify it

Reply:
Good catch, I'll check that out.

Reply:
I checked, it's root:root and 644 like the rest of the services.

Reply:
Just to be sure, the ownership is also root:root?